lind_syscall/crt1/crt1.c

```diff
@@ -1,39 +1,117 @@
-#include <wasi/api.h>
 #include <stdlib.h>
 #include <sysexits.h>
 
-int32_t __imported_wasi_snapshot_preview1_args_sizes_get(int32_t arg0, int32_t arg1) __attribute__((
+// char** __wasilibc_environ;
+// weak_alias(__wasilibc_environ, _environ);
+// weak_alias(__wasilibc_environ, environ);
+extern char** environ;
+
+static char *empty_environ[1] = { NULL };
+
+signed int __imported_wasi_snapshot_preview1_args_sizes_get(signed int arg0, signed int arg1) __attribute__((
     __import_module__("wasi_snapshot_preview1"),
     __import_name__("args_sizes_get")
 ));
 
-__wasi_errno_t __wasi_args_sizes_get(
-    __wasi_size_t *retptr0,
-    __wasi_size_t *retptr1
+unsigned short __wasi_args_sizes_get(
+    unsigned long *retptr0,
+    unsigned long *retptr1
 ){
-    int32_t ret = __imported_wasi_snapshot_preview1_args_sizes_get((int32_t) retptr0, (int32_t) retptr1);
-    return (uint16_t) ret;
+    signed int ret = __imported_wasi_snapshot_preview1_args_sizes_get((signed int) retptr0, (signed int) retptr1);
```

Comment:
Why are we changing the type definitions?  If the symbol isn't defined, isn't it easier to define it?